tools/UpdateModules.ps1

```diff
@@ -26,34 +26,151 @@ function Create-ModulePsm1
      $file = Get-Item -Path $templateOutputPath
   }
 }
 
+function Find-DefaultResourceGroupCmdlets
+{
+    [CmdletBinding()]
+    param(
+        [bool]$AddDefaultParameters,
+        [Hashtable]$ModuleMetadata,
+        [string]$ModulePath
+    )
+    PROCESS
+    {
+        if ($AddDefaultParameters) 
+        {
+        $nestedModules = $ModuleMetadata.NestedModules
+        $AllCmdlets = @()
+        $nestedModules | ForEach-Object {
+            $dllPath = Join-Path -Path $ModulePath -ChildPath $_
+            $Assembly = [Reflection.Assembly]::LoadFrom($dllPath)
+            $dllCmdlets = $Assembly.GetTypes() | Where-Object {$_.CustomAttributes.AttributeType.Name -contains "CmdletAttribute"}
+            $AllCmdlets += $dllCmdlets
+        }
+        
+        $FilteredCommands = $AllCmdlets | Where-Object {Test-RequiredParameter -Cmdlet $_}
+    
+        if ($FilteredCommands.Length -eq 0) {
+            $contructedCommands = "@()"
+        }
+        else {
+            $contructedCommands = "@("
+            $FilteredCommands | ForEach-Object {
+                $contructedCommands += "'" + $_.GetCustomAttributes("System.Management.Automation.CmdletAttribute").VerbName + "-" + $_.GetCustomAttributes("System.Management.Automation.CmdletAttribute").NounName + ":ResourceGroupName" + "',"
+            }
+            $contructedCommands = $contructedCommands -replace ".$",")"
+        }
+    
+        return $contructedCommands
+        }
+
+        else {
+        return "@()"
+        }
+    }
+}
+
+function Test-RequiredParameter
+{
+    [CmdletBinding()]
+    param(
+        [Object]$Cmdlet
+    )
+
+    PROCESS
+    {
+        $rgParameter = $Cmdlet.GetProperties() | Where-Object {$_.Name -eq "ResourceGroupName"}
+        if ($rgParameter -ne $null) {
+            $parameterSets = $rgParameter.CustomAttributes | Where-Object {$_.AttributeType.Name -eq "ParameterAttribute"}
+            $isMandatory = $true
+            $parameterSets | ForEach-Object {
+                $hasParameterSet = $_.NamedArguments | Where-Object {$_.MemberName -eq "ParameterSetName"}
+                $MandatoryParam = $_.NamedArguments | Where-Object {$_.MemberName -eq "Mandatory"}
+                if (($hasParameterSet -ne $null) -or (!$MandatoryParam.TypedValue.Value)) {
+                    $isMandatory = $false
+                }
+            }
+            if ($isMandatory) {
+                return $true
+            }
+        }
+        
+        return $false
+    }
+}
+
+function Create-MinimumVersionEntry
+{
+    [CmdletBinding()]
+    param(
+        [string]$ModuleName,
+        [string]$MinimumVersion
+    )
+
+    PROCESS
+    {
+        return "`$module = Get-Module $ModuleName `
+if (`$module -ne `$null -and `$module.Version.ToString().CompareTo(`"$MinimumVersion`") -lt 0) `
+{ `
+    Write-Warning `"A later version of $ModuleName was found to be imported already. Please see <link> for more details.`" `
```

Comment:
I think thsi is a bit out oif date with the latest in Cormac's PR.  Please pull the latest

Reply:
done